smarts/core/mission_planner.py

```diff
@@ -291,6 +292,16 @@ def uturn_waypoints(self, sim, pose: Pose, vehicle):
         current_edge = self._road_network.edge_by_lane_id(wp.lane_id)
         if not start_edge.oncoming_edges:
             return []
+        if self._task_is_triggered is False:
+            self._uturn_initial_heading = pose.heading
+
+        vehicle_heading_vec = radians_to_vec(pose.heading)
+        initial_heading_vec = radians_to_vec(self._uturn_initial_heading)
+
+        heading_diff = np.dot(vehicle_heading_vec, initial_heading_vec)
+
+        if heading_diff < -0.97:
```

Comment:
I could suggest using some ratio of `pi`.
```suggestion
        if heading_diff < -math.pi * 0.31:
```